src/service/tar_git

```diff
@@ -990,12 +997,29 @@ rpm_pkg () {
 }
 
 run_android_repo_service () {
-    case "$SVC_NAME" in
-	github) SVC_TAG=github;;
-	merproject) SVC_TAG=mer;;
-	jollamobile) SVC_TAG=jolla;;
-	*) echo "Unknown service '$SVC_NAME'"; exit 1;;
-    esac
+  if [[ ! -f $REPOCONFIG ]] ; then
+    echo "Missing repo config file $REPOCONFIG"
+    exit 1
+  fi
+  SVC_TAG=
+  while read svc_tag svc_url
+  do
+    svc_url="${svc_url#*://}" # remove scheme
+    svc_url="${svc_url%/*}"   # remove path
+    svc_url="${svc_url##*@}"  # remove username
+    svc_url="${svc_url%:*}"   # remove port
+    svc_url="${svc_url%.*}"   # remove trailing .something
+    svc_name="${svc_url#*.}"  # remove leading something.
+    if [[ "$svc_name" == "$SVC_NAME" ]]; then
+      SVC_TAG=svc_tag
+      break
+    fi
+  done < $REPOCONFIG
+  if [ -z "$SVC_TAG" ]; then
+    echo "Sorry, $SVC_REPO is not whitelisted. please contact ask your OBS administrator to add it to the repo config file $REPOCONFIG if you need it."
+    exit 1
+  fi
+
```

Comment:
tar_git has to create a file for repo to consume.

It has a git url.
repo takes a service name

This converts from a git url to a service name using the same config that repo uses to convert from a service name to a git url.